test/integration/resourcemanager/managedresource/reconciler_test.go

```diff
@@ -136,6 +142,55 @@ var _ = Describe("ManagedResource controller tests", func() {
 					return condition != nil && condition.Status == gardencorev1beta1.ConditionFalse
 				}, 30*time.Second, time.Second).Should(BeTrue())
 			})
+
+			It("should correctly set the condition ResourceApplied to Progressing", func() {
+				// finalizer is added so this resource can not be deleted once we remove it from the MangedResource
+				// referenced secret meanwhile adding a new resource in the refrenced secret set the ResourceApplied
+				// condition to progressing and stuck at this until we remove finalizer from the old resource
```

Comment:
```suggestion
				// this finalizer is added to prolong the deletion of the resource so that we can
				// observe the controller successfully setting ResourceApplied condition to Progressing
```